sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -122,6 +122,37 @@ object JdbcUtils extends Logging {
     }
   }
 
+  /**
+   * Runs a custom query against a table from the JDBC database.
+   */
+  def runQuery(conn: Connection, actions: String, options: JDBCOptions): Unit = {
+    val autoCommit = conn.getAutoCommit
+    conn.setAutoCommit(false)
```

Comment:
@moomindani Typically in my experience, in traditional databases, we do the pre and post trigger actions in the same transaction as the main action. I was saying that in spark, it's probably not possible as the main action could be done in a distributed fashion. So if we do this, we probably have to document it telling our users that, when a post action fails, we still can have the pre and post action committed.

Reply:
Makes sense. I have updated the document to include this point.